src/SixLabors.Fonts/TextLayout.cs

```diff
@@ -148,32 +146,31 @@ public IReadOnlyList<GlyphLayout> GenerateLayout(ReadOnlySpan<char> text, Render
                     }
                 }
 
-                if ((options.WrappingWidth > 0 && nextWrappableLocation == i) || nextWrappableRequired)
+                if ((options.WrappingWidth > 0 && nextWrappableLocation == index) || nextWrappableRequired)
                 {
                     // keep a record of where to wrap text and ensure that no line starts with white space
                     for (int j = layout.Count - 1; j >= 0; j--)
                     {
-                        if (!layout[j].IsWhiteSpace)
+                        if (!layout[j].IsWhiteSpace())
                         {
                             lastWrappableLocation = j + 1;
                             break;
                         }
                     }
                 }
 
-                if (nextWrappableLocation == i)
+                if (nextWrappableLocation == index && lineBreaker.MoveNext())
                 {
-                    if (lineBreaker.TryGetNextBreak(out b))
-                    {
-                        nextWrappableLocation = b.PositionWrap - 1;
-                        nextWrappableRequired = b.Required;
-                    }
+                    LineBreak b = lineBreaker.Current;
+                    nextWrappableLocation = b.PositionWrap - 1;
+                    nextWrappableRequired = b.Required;
                 }
 
                 float glyphWidth = glyph.AdvanceWidth * spanStyle.PointSize / scale;
                 float glyphHeight = glyph.Height * spanStyle.PointSize / scale;
 
-                if (codepoints[i] != '\r' && codepoints[i] != '\n' && codepoints[i] != '\t' && codepoints[i] != ' ')
+                // TODO: !CodePoint.IsWhitespace(codePoint) && ! CodePoint.IsBreak(codePoint) ??
+                if (codePoint.Value != '\r' && codePoint.Value != '\n' && codePoint.Value != '\t' && codePoint.Value != ' ')
```

Comment:
@tocsoft I'm curious about the explicit char rules here. This seems like something we can determine with our new Unicode tooling.

Reply:
Fixing via https://github.com/SixLabors/ImageSharp.Drawing/issues/131